my_enumerable_module.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+# Extending Enumerable module with replicated methods for the following:
+# each, each_with_index, select, all?, none?, any?, count, map and inject
+#
+module Enumerable
+  def my_each
+    arr = self # this is to prevent a redundant-use-of-self error
+    0.upto(arr.length - 1) do |i|
+      yield(arr[i])
+    end
+  end
+
+  def my_each_with_index
+    arr = self # this is to prevent a redundant-use-of-self error
+    0.upto(arr.length - 1) do |i|
+      yield(arr[i], i)
+    end
+  end
+
+  def my_select
+    arr = self # this is to prevent a redundant-use-of-self error
+    result = []
+    0.upto(arr.length - 1) do |i|
+      result << arr[i] if yield(arr[i])
+    end
+    result
+  end
+
+  def my_all?
+    arr = self # this is to prevent a redundant-use-of-self error
+    response = true
+    0.upto(arr.length - 1) do |i|
+      response = false unless yield(arr[i])
+    end
+    response
+  end
+
+  def my_any?
+    arr = self # this is to prevent a redundant-use-of-self error
+    response = false
+    0.upto(arr.length - 1) do |i|
+      response = true if yield(arr[i])
+    end
+    response
+  end
+
+  def my_none?
+    arr = self # this is to prevent a redundant-use-of-self error
+    response = true
+    0.upto(arr.length - 1) do |i|
+      response = false if yield(arr[i])
+    end
+    response
+  end
+
+  def my_count(single_value = nil)
+    arr = self
+    counter = 0
+    if single_value
+      arr.my_each { |val| counter += 1 if val == single_value }
+    else
+      arr.my_each { |val| counter += 1 if yield(val) }
+    end
+    counter
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns the number of items in enum through enumeration

Reply:
Corrected